src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -2126,6 +2126,13 @@ JRT_ENTRY_PROF(void, Deoptimization, uncommon_trap_inner, Deoptimization::uncomm
       Events::log_deopt_message(current, "Uncommon trap: reason=%s action=%s pc=" INTPTR_FORMAT " method=%s @ %d %s",
                                 reason_name, reason_action, pc,
                                 tm->name_and_sig_as_C_string(), trap_bci, nm->compiler_name());
+
+      if (nm->preloaded() && (action != Action_none)) {
```

Comment:
At some point, I added SCC-related info to `TraceDeoptimization` output [1], but it turns out I didn't upstream it. (I had a script which parsed the log and aggregated information about deopts in archived code.) 

Maybe it's better to upstream it instead of introducing such adhoc logging.

[1] https://github.com/iwanowww/leyden/blame/premain-precompile/src/hotspot/share/runtime/deoptimization.cpp#L2151

Reply:
But speaking of monitoring for uncommon traps in generated code: the ultimate goal may be to completely eliminate uncommon traps in preloaded code. In that respect, it's more interesting to monitor for all uncommon traps issued during compilation rather than those hit at runtime. 

For C2 it's `GraphKit::uncommon_trap()`, so I suggest to move the code there. Also, additional information about compilation (id) and location (jvms info) is useful to pin down the root cause.

Reply:
Yeah. I added logging at `GraphKit::uncommon_trap`. That logging is super-noisy. So I left the tracing code that warns when we trigger the actual deopt by stepping on such trap.